service/src/test/java/bio/terra/pipelines/dependencies/rawls/RawlsServiceTest.java

```diff
@@ -24,13 +33,159 @@ class RawlsServiceTest extends BaseEmbeddedDbTest {
+
+    rawlsService = spy(new RawlsService(rawlsClient, template));
+
+    doReturn(submissionsApi).when(rawlsClient).getSubmissionsApi(any());
+
+    assertEquals(expectedResponse, rawlsService.submitWorkflow(any(), any(), any(), any()));
+  }
+
+  @Test
+  void socketExceptionRetriesEventuallyFail() throws Exception {
+    SubmissionReport expectedResponse =
+        new SubmissionReport().status("status").submissionId(UUID.randomUUID().toString());
+
+    rawlsClient = mock(RawlsClient.class);
+    SubmissionsApi submissionsApi = mock(SubmissionsApi.class);
+    when(submissionsApi.createSubmission(any(), any(), any()))
+        .thenAnswer(errorAnswer)
+        .thenAnswer(errorAnswer)
+        .thenAnswer(errorAnswer)
+        .thenReturn(expectedResponse);
+
+    rawlsService = spy(new RawlsService(rawlsClient, template));
+
+    doReturn(submissionsApi).when(rawlsClient).getSubmissionsApi(any());
+
+    assertThrows(
+        SocketTimeoutException.class,
+        () -> {
+          rawlsService.submitWorkflow(any(), any(), any(), any());
+        });
+  }
+
+  @Test
+  void apiExceptionsDoNotRetry() throws Exception {
+    SubmissionReport expectedResponse =
+        new SubmissionReport().status("status").submissionId(UUID.randomUUID().toString());
+
+    ApiException expectedException = new ApiException(400, "Bad Rawls");
+
+    rawlsClient = mock(RawlsClient.class);
+    SubmissionsApi submissionsApi = mock(SubmissionsApi.class);
+    when(submissionsApi.createSubmission(any(), any(), any()))
+        .thenThrow(expectedException)
+        .thenReturn(expectedResponse);
+
+    rawlsService = spy(new RawlsService(rawlsClient, template));
+
+    doReturn(submissionsApi).when(rawlsClient).getSubmissionsApi(any());
+
+    RawlsServiceApiException thrown =
+        assertThrows(
+            RawlsServiceApiException.class,
+            () -> {
+              rawlsService.submitWorkflow(any(), any(), any(), any());
+            });
+    assertEquals(expectedException, thrown.getCause());
+  }
+
+  @Test
+  void submitWorkflow() throws Exception {
+    SubmissionReport expectedResponse =
+        new SubmissionReport().status("status").submissionId(UUID.randomUUID().toString());
+
+    rawlsClient = mock(RawlsClient.class);
+    SubmissionsApi submissionsApi = mock(SubmissionsApi.class);
+    when(submissionsApi.createSubmission(any(), any(), any())).thenReturn(expectedResponse);
+
+    rawlsService = spy(new RawlsService(rawlsClient, template));
+
+    doReturn(submissionsApi).when(rawlsClient).getSubmissionsApi(any());
+
+    assertEquals(expectedResponse, rawlsService.submitWorkflow(any(), any(), any(), any()));
+  }
+
+  @Test
+  void getSubmissionStatus() throws Exception {
+    UUID expectedUUID = UUID.randomUUID();
+    Submission expectedResponse =
+        new Submission()
+            .status(SubmissionStatus.SUBMITTING)
+            .submissionId(expectedUUID.toString())
+            .cost(3.45F);
```

Comment:
oh cool that cost comes back. probably not in any timely way for us to do anything with it, right?